easybuild/tools/module_generator.py

```diff
@@ -213,25 +213,20 @@ def _filter_paths(self, key, paths):
             return paths
 
         added_paths = self.added_paths_per_key.setdefault(key, set())
-        # Coerce any iterable/generator into a list
-        if not isinstance(paths, list):
-            paths = list(paths)
```

Comment:
I'd make this an error such that e.g. passing a string doesn't result in garbage